src/base/misc/fatfs.c

```diff
@@ -206,11 +219,6 @@ void fatfs_init(struct disk *dp)
   f->reserved_secs = 1;
   f->hidden_secs = dp->start;
   f->fats = 2;
-  if (f->fat_type == FAT_TYPE_FAT12) {
-    f->fat_secs = ((f->total_secs / f->cluster_secs + 2) * 3 + 0x3ff) >> 10;
-  } else {
-    f->fat_secs = ((f->total_secs / f->cluster_secs + 2) * 2 + 0x1ff) >> 9;
-  }
```

Comment:
Because then there are excessive parameters.
If fat_secs is not equal to the ebove formula, then
some parameter is wrong.
But I think fat_secs is indeed the "primary" parameter.
So I suppose your patch is fine and the "total_secs"
should instead be calculated from fat_secs (it is currently
calculated from CHS, which looks wrong).
So if you hard set fat_secs, then please make total_secs
calculated. For example if the partition size is not aligned
to the cluster size, than you can't directly move it to total_secs,
right?